### ISSUE TITLE
Is LightCleaner still necessary?

### ISSUE THREAD

**Issue by PaBox** [IntellectualSites/PlotSquared#3514]
### Server Implementation

Paper

### Server Version

1.16.5

### Describe the bug

When generating new chunks with a custom template that includes lighting-blocks, the lighting is not updated after the chunk is placed the first time. Instead LightCleaner was advertised in the past. 

The problem is, that LightCleaner goes into an infinite loop of calculating light for unloaded chunks and this is costing a lot of performance. The author of that plugin said, that the problem relies elsewhere and it is not his fault, so I wonder where this problem comes from.

### To Reproduce

1. Use Template with lighting blocks, for example 'Bridge.template'
2. Generate Chunks
3. See no lighting

### Expected behaviour

PlotSquared should handle the lighting itself.

### Plot Debugpaste

https://athion.net/ISPaster/paste/view/ca36ed467388451b9ef4e047cf52e247

### PlotSquared Version

PlotSquared version 6.3.0

### Checklist

- [X] I have included a Plot debugpaste.
- [X] I am using the newest build from https://www.spigotmc.org/resources/77506/ and the issue still persists.

**Comment by PierreSchwang** [comment]
> PlotSquared version 6.3.0
Try with a more recent build and see if the issue persists.

**Comment by PierreSchwang** [comment]
Cannot reproduce with PlotSquared 6.5.1 and FAWE 2.0.2

**Comment by PaBox** [comment]
As you can see, On 1.16.5 I am using the newest versions I can use, being 6.3.0 PlotSquared, as 6.4.0 requires JAVA 17. Same with FAWE. Both versions you named are not compatible to 1.16.5...

**Comment by PierreSchwang** [comment]
Paper 1.16.5 is compatible with Java 17
> If you are using a version prior to 1.17, add -DPaper.IgnoreJavaVersion=true to your JVM flags and your server skips the Java version check.

**Comment by PaBox** [comment]
I searched their comments on the patch and they dont state it being compatible, as this is a clear workaround and they cant GUARANTEE it to work. But I just saw that on the main page of your plugin it states to use 6.0.0 for 1.16.5...? Is that just old or am I missing something... Still thank you for that hint at that point, it is not documented at that point if I see it right...

**Comment by PierreSchwang** [comment]
The version check is just some dumb stuff which minecraft introduced, who knows why - i've had no issue nor received any reported issues when running paper 1.16 on java 17

**Comment by PaBox** [comment]
I just tried this and used PS Version 6.5.1 with FAWE 2.0.2-103. This is still a bug.

**Comment by PaBox** [comment]
And I do not think that this is a FAWE issue, as the World Generator is the problem here or not... I am talking about road generation here...

**Comment by PierreSchwang** [comment]
Still works perfectly fine 
![grafik](https://user-images.githubusercontent.com/27054324/155602792-7e73ccaf-065f-411f-8506-33c062285bc0.png)
![grafik](https://user-images.githubusercontent.com/27054324/155602834-ea7eb57a-ee60-44ef-9bf6-2675d62b4c5b.png)

**Comment by PaBox** [comment]
Yes, the debug version of the command which places one corner also comes with block updates, which is quite annoying when working with floating blocks. But if you use the template and fly out a little bit further so chunks actually generate with the world generator, the light sources are still NOT updated. 

What I have done is also enable the in-built options for Lighting of both FAWE and PlotSquared again, as LightCleaner could have interfered at that point. Should I create a new template? Another Debugpaste maybe with the current setup so one can compare? (Though I think you are using default settings)

I will provide you with a screenshot:
![2022-02-23_22 38 38](https://user-images.githubusercontent.com/13438904/155626422-4ea68466-dc9b-45d4-824d-75a4ba44857a.png)

**Comment by PaBox** [comment]
Is this repeatable? Otherwise I will create a clean server on 1.18 and see if it works and create another Issue (then close this). But this is a persistent problem in 1.16.5 if blocks are placed using the world generator (not the debug command) and if it is the only way around we might have to update to 1.18. But at the moment not even that is certain and I do not have much time for testing at the moment, so I might add something to this thread when a fix is already implemented. Watching the progress on Github though in the meantime.

**Comment by PierreSchwang** [comment]
I see, I'm now able to replicate that

**Comment by PaBox** [comment]
Would you remove the label 'Cannot replicate' then? This is an issue which has persisted for quite some time now (look at any youtube video using the bridge template available on youtube ever!, even the spigot page photos have that problem since at least v4 :)

![grafik](https://user-images.githubusercontent.com/13438904/155905393-eccdd610-b8f1-43df-92e4-319c88298a73.png)

https://www.spigotmc.org/resources/plotsquared-v4-v6-out-now.1177/

The problem is that the world generator does no light updates itself and does not hook into the settings of plotsquared and neither into the lighting system of FAWE (which could trigger Light Cleaner properly) - but FAWE cant generate new chunks like a world generator as far as I am aware of it. 

As this has bothered quite a lot of people over all this time (while many dont notice it I suspect), I wonder, why this has not been recognized before by the devs - I will take a look at the code, maybe I can find the whole world-generation specific part, see whats wrong and what could be done better.

Cheerio for now.

**Comment by NotMyFault** [comment]
> I don't have the permissions to do that anymore - You'd have to wait until a code owner sees this.

Should be resolved by now.

**Comment by PierreSchwang** [comment]
I'm not quite sure if that's a FAWE issue or a PS issue yet - I'd have to look through the code for that

**Comment by dordsor21** [comment]
There's no API in bukkit/spigot(/paper that I'm aware of) that allowed for lighting to be "generated" with lighting blocks placed. You can also quite easily see many lighting issues during generation from Minecraft itself from lava lakes above ground, and sometimes underground.

**Comment by PaBox** [comment]
Is this an answer or just and information? Minecraft fixed most of its lighting bugs with 1.14.4 and I havent been seeing those shadows for a long time actually. My lava lakes are fine... Why arent the schematics pasted in with fawe?

**Comment by PaBox** [comment]
This does work now somewhat, but I suspect it being updated in a wrong order and or claiming and clearing the plot doesnt trigger the light level recalculation. Its also most noticable where 2 schematics meet each other - so maybe update the neighboring chunks as well? I dont really know what could be the issue here:
![2023-03-16_02 42 12](https://user-images.githubusercontent.com/13438904/225488119-db6e2a36-91c1-4216-9434-9d101d795be8.png)
With Cleanlight I get the correct output: 
![2023-03-16_02 42 36](https://user-images.githubusercontent.com/13438904/225488444-6f9e5125-2038-4e89-9074-9d4a56dfa5b5.png)
Its not really much of an issue right now, but maybe you can take a look at it if you got some time :)
